docarray/array/stacked/column_storage.py

```diff
@@ -0,0 +1,196 @@
+from collections import ChainMap
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Dict,
+    Iterable,
+    List,
+    MutableMapping,
+    Sequence,
+    Type,
+    TypeVar,
+    Union,
+    cast,
+)
+
+from docarray.array.array.array import DocumentArray
+from docarray.base_document import BaseDocument
+from docarray.typing.tensor.abstract_tensor import AbstractTensor
+from docarray.utils._typing import is_tensor_union
+from docarray.utils.misc import is_tf_available
+
+if TYPE_CHECKING:
+    from docarray.array.stacked.array_stacked import DocumentArrayStacked
+
+tf_available = is_tf_available()
+if tf_available:
+    import tensorflow as tf  # type: ignore
+
+    from docarray.typing import TensorFlowTensor
+else:
+    TensorFlowTensor = None  # type: ignore
+
+IndexIterType = Union[slice, Iterable[int], Iterable[bool], None]
+
+
+T = TypeVar('T', bound='ColumnStorage')
+
+
+class ColumnStorage:
+
+    document_type: Type[BaseDocument]
+
+    tensor_columns: Dict[str, AbstractTensor]
+    doc_columns: Dict[str, 'DocumentArrayStacked']
+    da_columns: Dict[str, List['DocumentArrayStacked']]
+    any_columns: Dict[str, List]
```

Comment:
why is `da_columns` separate? It's just a list anyways, right?

Reply:
We need special treatment for the da inside da. So gut feeling is that we should make it a different variable